### PR TITLE
refactor(events): replace underscore result lookup

### PR DESCRIPTION
**Pull request by paulfalgout**
Related #241

## Linked issue

- Related #241

## What

- Add a private literal-key `getValue` helper for Marionette configuration lookup.
- Migrate `modelEvents` and `collectionEvents` resolution away from Underscore `result`.
- Add differential utility coverage and receiver/zero-argument integration assertions, then regenerate affected main distribution artifacts.

## Why

Marionette has many single-property `result` calls but no deep-path callers. Owning the exact literal-key contract once avoids repeating subtle lookup and invocation logic as the remaining Underscore consumers migrate.

## Agent-development failure addressed

- Gives agents one explicit, tested abstraction for configuration values that may be properties or functions.
- Documents inherited lookup, undefined-only fallback, receiver binding, zero arguments, nullish inputs, proxies, getters, and exception propagation rather than leaving those semantics implicit in Underscore.

## Public behavior contract

- Canonical behavior after this PR: read the literal property once, use the fallback only for `undefined`, and invoke callable values with the source object as receiver and no arguments.
- Superseded behavior removed, if any: direct use of Underscore `result` in entity-event delegation.
- Compatibility exception and removal condition, if any: no transition path is added. Unused Underscore deep-path and mutable `_.toPath` coupling are intentionally not reproduced because every repository caller uses a literal property key.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: the size/resource contract passes; the direct production Underscore importer count drops from 16 to 15.
- Startup/hot path: the helper is a small synchronous property lookup and conditional call, with no framework registration or initialization cost.
- Allocations/retention: no per-instance state or retained objects are added; ordinary non-callable lookups allocate nothing.

## Scope

- Affects `modelEvents` and `collectionEvents` configuration resolution plus one private utility intended for later literal-key migrations.
- Does not migrate other `result` call sites, change event binding order, or alter rendering/application behavior.

## Deployment Impact

- No website, package, tag, release, or deployment is performed by this PR.
- No consumer redeploy is required beyond adopting a future Marionette package containing this change.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm ci --no-audit --no-fund
npm exec vitest -- run test/unit/utils/get-value.spec.js test/unit/mixins/delegate-entity-events.spec.js --reporter=dot
npm run lint:ci
npm run build
npm test -- --reporter=dot
npm run test:source
npm run test:dist
npm run test:fixtures
npm run size
npm run test:performance
npm run performance:timing
git diff --check
```

Results: focused tests passed 20/20; full tests passed 1,442/1,442 across 76 files; source, distribution, packed fixture, lint, build, size/resource, 92 performance-contract, hosted timing, and diff validation passed. Independent exact-head review passed 48/48 focused assertions with no findings.

## Package and browser impact

- Entry points or install fixtures affected: main ESM, CJS, UMD, and minified artifacts are regenerated; packed ESM/CJS/type fixtures pass.
- Browser contracts affected: none; the helper uses baseline property access, nullish comparison, `typeof`, and `Function.prototype.call` behavior already required by Marionette.
- Production/dev/test module-graph impact: one production module stops importing Underscore; other production consumers still retain it.

## Rollback or deprecation

- Revert this single commit and regenerate the main distribution artifacts; no data migration or compatibility bridge is involved.

## Review notes

- The helper deliberately accepts a literal key rather than cloning Underscore deep-path support that Marionette does not use.
- Getter, proxy, callable-proxy, class-constructor, revoked-proxy, inherited-property, fallback, receiver, argument-count, and error propagation cases are covered differentially against Underscore.

---
## Summary by cubic
Replaces Underscore's `result` calls for `modelEvents` and `collectionEvents` with a small private `getValue` helper, continuing the Underscore migration from #241. The helper reads a literal key once, falls back only for `undefined`, and invokes callable values with the source object as receiver and no arguments.

**Refactors**
- Adds `utils/get-value.js` with differential tests against `_.result`.
- Regenerates the main `dist` artifacts; one production module stops importing Underscore.
- Deliberately omits Underscore's deep-path support since no callers use it.

<sup>Written for commit 0d185fbd38d685cbf5551ddb893992e2fb6d4831. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/278?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved delegated model and collection event handling.
  * Event configuration providers are now invoked consistently with the correct context and no arguments.
  * Preserved expected fallback behavior when event configuration values are unavailable.

* **Tests**
  * Added comprehensive coverage for value resolution, fallbacks, callable values, inherited properties, proxies, getters, and error handling.

### CODE REVIEW

**Comment by paulfalgout** [comment]
## Code-review approval — exact head `0d8ea976`

**Approved.** CI green, +273 B, 17,327 remaining. Production Underscore importers **17 → 16**.

Still source-only review — `node_modules` is empty in my copy — but like #277 this is a case where reading is nearly as good as running, because the helper is a line-for-line reproduction of `_.result`'s single-key path:

```
getValue                                          _.result (single key)
object == null ? undefined : object[property]  ≡  obj == null ? void 0 : obj[path[i]]
value === undefined ? fallback : value          ≡  if (prop === void 0) prop = fallback
typeof rv === 'function' ? rv.call(object) : rv ≡  isFunction(prop) ? prop.call(obj) : prop
```

Every semantic in your list falls out of that correspondence rather than needing separate handling: inherited lookup because it's a plain property read, **undefined-only** fallback because the test is `=== undefined` and not nullish, receiver binding and zero arguments from `.call(object)`, getter/proxy traps firing exactly once, and exceptions propagating because nothing catches. A function *fallback* is invoked too, which is easy to miss and matches `_.result`.

**I verified the deep-path narrowing rather than taking it on trust.** Grepping every `result(` call in production source: all are single-key, none pass an array path, so dropping `_.result`'s array support removes capability nothing uses. The three-argument fallback form does have a live caller — `mixins/view.js:197`, `result(this, 'childViewEventPrefix', defaultPrefix)` — and `getValue` supports it identically.

**Building this once before migrating the rest is the right sequencing.** There are ~20 remaining `result(` sites across `behaviors`, `radio`, `ui`, `view-events`, `view`, `template-render`, and `common`. Getting the contract exact here, with the differential coverage attached, means those migrations become mechanical substitutions rather than twenty separate opportunities to subtly re-derive `undefined`-versus-`null` or receiver binding. +273 B for the helper plus tests is front-loaded cost against that.

<sub>Reviewed with Claude Code.</sub>

File: test/unit/utils/get-value.spec.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import _ from 'underscore';
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/utils/get-value.spec.js:1]: Handled in `test/unit/utils/get-value.spec.js` with standalone assertions for one read, receiver binding, zero arguments, and undefined-only fallback.

**Comment by paulfalgout** [comment]
**Approval carries** to `0d185fbd` (the SHA I fetched). Tests-only — `dist/`, `utils/`, `mixins/`, and `modules/` are byte-identical to `0d8ea976`, so the runtime I reviewed is unchanged. CI green.

Both additions pin the two semantics I singled out as easiest to lose, which is the right instinct given ~20 call sites will migrate onto this helper:

The read-once test uses a getter to count property access and asserts `reads === 1`. That catches a genuinely easy mistake — a reimplementation that reads `object[property]` once to test and again to use would still pass every value-based assertion while firing a getter or Proxy trap twice. Capturing the receiver and `args` in the same test covers `.call(object)` with zero arguments in one pass.

The fallback test is the one I'd most want present: `null`, `false`, `0`, and `''` all decline the fallback while `undefined` (explicit and missing alike) takes it. `_.result`'s `=== void 0` check is the detail a "sensible" rewrite most often converts to a nullish or falsy test, and it would be invisible until someone's `modelEvents` legitimately resolved to `null`.

Worth noting the explicit-`undefined` and absent-property cases are both covered rather than just one — they take the same branch here, but asserting both means a future change that starts distinguishing `'value' in object` gets caught.

<sub>Reviewed with Claude Code.</sub>
